fish-pepper/jolokia/fp-files/jolokia-opts

```diff
@@ -13,65 +10,83 @@ is_in_jolokia_opts() {
+  jolokia_property_file="$1"
+
+  # Setup Jolokia to accept basic auth, using a randomly generated password that is stored
+  # in the container in the ${DEPLOYMENTS_DIR}/jolokia.pw file.
+  if [ "$AB_JOLOKIA_PASSWORD_RANDOM" == "true" ]; then
+    pw_file="/opt/jolokia/etc/jolokia.pw"
+    if [ -f "${pw_file}" ] ; then
+      AB_JOLOKIA_PASSWORD=`cat "${pw_file}"`
+    else
+      AB_JOLOKIA_PASSWORD=`tr -cd '[:alnum:]' < /dev/urandom | fold -w30 | head -n1`
+      touch "${pw_file}"
+      chmod 600 "${pw_file}"    
+      cat > "${pw_file}" <<EOF
+$AB_JOLOKIA_PASSWORD
+EOF
+    fi
+    export AB_JOLOKIA_PASSWORD
+  fi
+
+  touch "${jolokia_property_file}"
+  chmod 600 "${jolokia_property_file}"
+  cat > "${jolokia_property_file}" <<EOF
+$(get_jolokia_properties)  
+EOF
+
+}
+
 if [ -z ${AB_JOLOKIA_OFF+x} ]; then
-   opts="-javaagent:$dir/jolokia.jar"
-   config=${AB_JOLOKIA_CONFIG:-$dir/jolokia.properties}
-   if [ -f "$config" ]; then
-      # Configuration takes precedence
-      opts="${opts}${sep}config=${config}"
-      sep=","
-      grep -q -e '^host' ${config} && host_in_config=1
-   fi
-   if [ -z ${AB_JOLOKIA_HOST+x} ] && [ -z ${host_in_config+x} ]; then
-      AB_JOLOKIA_HOST='0.0.0.0'
-   fi
-   if [ -n "$AB_JOLOKIA_HOST" ]; then
-      opts="${opts}${sep}host=${AB_JOLOKIA_HOST}"
-      sep=","
-   fi
-   if [ -n "$AB_JOLOKIA_PORT" ]; then
-      opts="${opts}${sep}port=${AB_JOLOKIA_PORT}"
-      sep=","
-   fi
-   if [ -n "$AB_JOLOKIA_USER" ]; then
-      opts="${opts}${sep}user=${AB_JOLOKIA_USER}"
-      sep=","
-   fi
-   if [ -n "$AB_JOLOKIA_PASSWORD" ]; then
-      opts="${opts}${sep}password=${AB_JOLOKIA_PASSWORD}"
-      sep=","
-   fi
-   if [ -n "$AB_JOLOKIA_HTTPS" ]; then
-      opts="${opts}${sep}protocol=https"
-      use_https=1
-      sep=","
-   fi
-   # Integration with OpenShift client cert auth
-   if [ -n "$AB_JOLOKIA_AUTH_OPENSHIFT" ]; then
-      auth_opts="useSslClientAuthentication=true,extraClientCheck=true"
-      if [ -z ${use_https+x} ]; then
-        auth_opts="${auth_opts},protocol=https"
-      fi
-      if [ $(is_in_jolokia_opts "caCert") != "yes" ]; then
-         auth_opts="${auth_opts},caCert=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
-      fi
-      if [ $(is_in_jolokia_opts "clientPrincipal") != "yes" ]; then
-         if [  x"${AB_JOLOKIA_AUTH_OPENSHIFT}" != x"${AB_JOLOKIA_AUTH_OPENSHIFT/=/}" ]; then
-            # Supposed to contain a principal name to check
-            auth_opts="${auth_opts},clientPrincipal=`echo ${AB_JOLOKIA_AUTH_OPENSHIFT} | sed -e 's/ /\\\\ /g'`"
-         else
-            auth_opts="${auth_opts},clientPrincipal=cn=system:master-proxy"
-         fi
-      fi
-      opts="${opts}${sep}${auth_opts}"
-      sep=","
-   fi
-   # Add extra opts to the end
-   if [ -n "$AB_JOLOKIA_OPTS" ]; then
-      opts="${opts}${sep}${AB_JOLOKIA_OPTS}"
-      sep=","
-   fi
-   if [ "x$sep" != 'x=' ] ; then
-     echo ${opts}
-   fi
+  if [ "$AB_JOLOKIA_CONFIG"=="" ]; then
```

Comment:
This will be always true since it is a single string. You need to add spaced around the operator. Next then `==` as operator does not exist for `test`.  You better you should use one of the standard idioms like `-z "$AB_JOLOKIA_CONFIG"` or the old school style which works since the 1980ies '`x"$AB_JOLOKIA_CONFIC" = x`

shell coding, we love it ;-)